common/src/main/kotlin/org/jetbrains/kotlinconf/presentation/DataRepository.kt

```diff
@@ -4,39 +4,17 @@ import org.jetbrains.kotlinconf.*
 import org.jetbrains.kotlinconf.data.*
 
 interface DataRepository {
-    val sessions: List<SessionModel>
-    val favorites: List<SessionModel>
-    val votes: List<Vote>
-
-    fun getSessionById(id: String): SessionModel
+    val sessions: List<SessionModel>?
+    val favorites: List<SessionModel>?
+    val votes: List<Vote>?
+    var onRefreshListeners: List<()->Unit>
+    val loggedIn: Boolean
+    var codePromptShown: Boolean
 
     suspend fun update()
-
-    /**
-     * Rating
-     */
-    fun getRating(sessionId: String): SessionRating
+    fun getRating(sessionId: String): SessionRating?
     suspend fun addRating(sessionId: String, rating: SessionRating)
     suspend fun removeRating(sessionId: String)
-
-    /**
-     * Favorites
-     */
-    fun isFavorite(sessionId: String): Boolean
     suspend fun setFavorite(sessionId: String, isFavorite: Boolean)
-
-
-    /**
-     * Code verification
-     */
     suspend fun verifyCode(code: VotingCode)
-
-    /**
-     * Native callback API
-     */
-    fun update(callback: NativeCallback<Unit>)
```

Comment:
The kotlin-native doesn't expose suspend functions to the iOS stubs.
So we should add alternative with callbacks for the iOS

Reply:
No, we don't. Now I use Presenters on iOS and I don't need to use repository from views (as it should be in MVP). Check out iOS. It is fully operational.